crates/sui-swarm/src/memory/swarm.rs

```diff
@@ -207,6 +210,15 @@ impl<R> SwarmBuilder<R> {
         self
     }
 
+    pub fn with_overload_threshold_config(
+        mut self,
+        overload_threshold_config: OverloadThresholdConfig,
+    ) -> Self {
+        assert!(self.network_config.is_none());
```

Comment:
ooc why must this be the case?

Reply:
Because overload_threshold_config is part of network config. If we are already customizing network config, we shouldn't overwrite it.